webflow-hacks/form-hacks/17/17.ts

```diff
@@ -0,0 +1,50 @@
+document.addEventListener('DOMContentLoaded', () => {
+  //assign data attributes to constants
+  const FORM_BUTTON_UP_SELECTOR = '[fs-hacks-element="hack17-counter-button-up"]';
+  const FORM_BUTTON_DOWN_SELECTOR = '[fs-hacks-element="hack17-counter-button-down"]';
+  const COUNTER_INPUT_SELECTOR = '[fs-hacks-element="hack17-counter-input"]';
+
+  /*
+In the video, the classes are referencing "Hack 16"
+This Hack launched as "Hack 17"
+Please note the change of classes from 16 to 17
+*/
+
+  // select elements with their respective data attributes
+  const counterInputs = document.querySelectorAll<HTMLSelectElement>(COUNTER_INPUT_SELECTOR);
+  if (!counterInputs) return;
+
+  const incrementButtons = document.querySelectorAll<HTMLSelectElement>(FORM_BUTTON_UP_SELECTOR);
+  if (!incrementButtons) return;
+
+  const decrementButtons = document.querySelectorAll<HTMLSelectElement>(FORM_BUTTON_DOWN_SELECTOR);
+  if (!decrementButtons) return;
+
+  // loop through each counter input
+  counterInputs.forEach((item, index) => {
+    // declare individual components
+    const counterInput = counterInputs[index];
+    const incrementButton = incrementButtons[index];
+    const decrementButton = decrementButtons[index];
```

Comment:
This assumes that the increment/decrement buttons will always be right next to the counter input.
We can't make these assumptions, people tend to build all kinds of layouts where the elements aren't necessary one to each other.

I see a few potential bugs here:
- The amount of increment/decrement buttons not being the same amount of counter inputs.
- The increment/decrement buttons' index not matching the related counter input's index.

Reply:
@alexiglesias93 then I would use different ids or selectors for each counter? or how do you recommend I change the code?

Reply:
with event.target or this ?

Reply:
Or you can use two attributes, one for the selector and another for the instance index:
`[attribute="selector"]`
[instance-attribute="1"]